### PR TITLE
feat(logout): Small improvements

### PR DESCRIPTION
**Pull request by krizzu**
## 📑 What does this PR do?

Addressing comments from #111 

# ✅ Checklist

- [X] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [X] All the tests have passed

### CODE REVIEW

File: ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponent.kt [hunk 1]
```diff
@@ -49,4 +48,10 @@ interface RootComponent {
     sealed interface DeepLink {
         object None : DeepLink
     }
+
+    sealed class UiAuthStatus {
```
Comment (crocsandcoffee) [inline ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponent.kt:52]: Please move this to the same file as the composable or to its own file. It doesn't need to be coupled with the component file namespace

Comment (crocsandcoffee) [reply ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponent.kt:52]: Also is unknown ever used?

File: ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt [hunk 2]
```diff
@@ -17,31 +17,23 @@ import kotlinx.coroutines.CoroutineScope
...
-interface RootComponentViewModel : InstanceKeeper.Instance {
-    val authState: StateFlow<AuthStatus?>
-}
-
-fun RootComponentViewModel(coroutineContext: CoroutineContext, getAuthStatus: GetAuthStatus): RootComponentViewModel =
-    RealRootComponentViewModel(coroutineContext, getAuthStatus)
-
-private class RealRootComponentViewModel(coroutineContext: CoroutineContext, getAuthStatus: GetAuthStatus) :
-    RootComponentViewModel {
+class RootComponentViewModel(coroutineContext: CoroutineContext, getAuthStatus: GetAuthStatus) :
```
Comment (crocsandcoffee) [inline ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt:26]: Nit: please format so params are on their own lines for clearer readability/style/formatting

File: ui/root/src/commonMain/kotlin/social/androiddev/root/splash/SplashContent.kt [hunk 3]
```diff
@@ -60,12 +60,12 @@ fun SplashContent(
         Text("Loading")
 
         LaunchedEffect(authStatus) {
-            if (authStatus is AuthStatus.Authorized) {
```
Comment (crocsandcoffee) [inline ui/root/src/commonMain/kotlin/social/androiddev/root/splash/SplashContent.kt:63]: We should always prefer exhaustive `when` over if/else chaining when using enum or sealed class

File: ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt [hunk 4]
```diff
@@ -17,31 +17,23 @@ import kotlinx.coroutines.CoroutineScope
...
+    InstanceKeeper.Instance {
     private val viewModelScope = CoroutineScope(coroutineContext + SupervisorJob())
 
-    private val _authState = MutableStateFlow<AuthStatus?>(null)
-    override val authState = _authState.asStateFlow()
+    private val _authState = MutableStateFlow<RootComponent.UiAuthStatus>(RootComponent.UiAuthStatus.Unknown)
```
Comment (crocsandcoffee) [inline ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt:30]: You can still simplify the `init` block and consolidate this private and public property using my suggestion from your other PR. Here's the link to my comment
https://github.com/AndroidDev-social/DodoForMastodon/pull/111#discussion_r1061676320

**Review by crocsandcoffee — COMMENTED** [review]
Thanks for the follow-up!

File: ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt (hunk 4 above)
Comment (krizzu) [reply ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt:30]: the `stateIn` call is suspend call, so can't do it that way

File: ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponent.kt (hunk 1 above)
Comment (krizzu) [reply ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponent.kt:52]: Sure thing. Yes, `unknown` is used as initial value

File: ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt (hunk 4 above)
Comment (crocsandcoffee) [reply ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt:30]: I'm not sure I'm following... Please see the doc
https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/state-in.html

Comment (crocsandcoffee) [reply ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt:30]: Your repo returns a Flow.
You should be able to do something similar to 
```kotlin
val state: StateFlow<State> = repo.getFlow().stateIn(scope, SharingStarted.Eagerly, State.LOADING)
```

File: ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponent.kt (hunk 1 above)
Comment (crocsandcoffee) [reply ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponent.kt:52]: Ah right, I missed that. Might be clearer to rename to UIAuthStatus.Loading or make nullable instead.

File: ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt (hunk 4 above)
Comment (krizzu) [reply ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt:30]: That'd be true, if `stateIn` would not be a suspend call - The docs mention the function signature (at the bottom).

> Starts the upstream flow in a given scope, suspends until the first value is emitted, and returns a hotStateFlow of future emissions

The function signature is 
```kotlin
suspend fun <T> Flow<T>.stateIn(scope: CoroutineScope): StateFlow<T>
```

File: ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponent.kt (hunk 1 above)
Comment (krizzu) [reply ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponent.kt:52]: I like `Loading` better 👍

File: ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt (hunk 2 above)
Comment (krizzu) [reply ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt:26]: Will do - too bad spotless can't do that 😞

File: ui/root/src/commonMain/kotlin/social/androiddev/root/splash/SplashContent.kt (hunk 3 above)
Comment (krizzu) [reply ui/root/src/commonMain/kotlin/social/androiddev/root/splash/SplashContent.kt:63]: I'm only interested in 2 cases out of 3, does the rule still apply?

Comment (crocsandcoffee) [reply ui/root/src/commonMain/kotlin/social/androiddev/root/splash/SplashContent.kt:63]: Still good practice, so you get the compiler warning/error if you ever add a new value to the enum/sealed class. You can always have it just be a no-op here for the ones you don't care about

File: ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt (hunk 4 above)
Comment (krizzu) [reply ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt:30]: Ah, perfect - That'd work!

**Comment by krizzu** [comment]
@crocsandcoffee Thanks for checking it out!

File: ui/root/src/commonMain/kotlin/social/androiddev/root/splash/SplashContent.kt [hunk 5]
```diff
@@ -60,13 +60,14 @@ fun SplashContent(
...
+            when (authStatus) {
                 // TODO(krzysztof): Do not navigate to timeline, until logout func is implemented
                 //                  https://github.com/AndroidDev-social/DodoForMastodon/issues/107
-                navigateToWelcome()
-                // navigateToTimeline()
-            } else if (authStatus is AuthStatus.Unauthorized) {
-                navigateToWelcome()
+                is UiAuthStatus.Authorized,
```
Comment (crocsandcoffee) [inline ui/root/src/commonMain/kotlin/social/androiddev/root/splash/SplashContent.kt:66]: Shouldn't you go to the timeline if you're authorized?

**Review by crocsandcoffee — APPROVED** [review]
Approving but plz confirm that last comment about navigating to timeline. Look good otherwise 👍

**Pull request merged**